### PR TITLE
Pro 3459 pin not working for international numbers

### PR DESCRIPTION
**Pull request by davejones74**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRO-3459

### Change description ###
PinController.java has been updated to change the generate pin endpoint for invites.
Exsiting invite endpoint /pin/{phoneNumber}
New invite endpoint /pin?phoneNumber={URLEncodedPhoneNumber} 

This change was required to address the issues with international numbers not appearing in the Business Service logs which was resulting in the DebugConsole showing 404 messages for anything with a '+' included in the number. With no reference in the logs, then there was no subsiquient call to send the SMS message.

This change affects the Probate PA Frontend application as well, so the existing endpoint for now /pin/{phoneNumber} has been retained to allow the frontend application to continue until the PRO-3459 branch is merged for the frontend application.

It should be noted that although this has been tested in development, the build process while successful for Checkout, Build, Test, Security Checks, Sonar Scan has suddently stopped - I'm assuming this is because of the ongoing environment issues in Azure.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/probate/functional/BusinessServicePinControllerTests.java [hunk 1]
```diff
@@ -27,16 +27,17 @@ public void testInviteFailure() {
     }
 
     private void validatePinSuccess(String phoneNumber) {
+
```
Comment (tomxelliott) [inline src/functionalTest/java/uk/gov/hmcts/probate/functional/BusinessServicePinControllerTests.java]: Please get rid of empty line

File: src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java [hunk 2]
```diff
@@ -11,15 +16,37 @@
...
-    @RequestMapping(path = "/pin/{phoneNumber}", method = RequestMethod.GET)
-    public String invite(@PathVariable String phoneNumber,
-                         @RequestHeader("Session-Id") String sessionId) throws NotificationClientException {
-
-        LOGGER.info("Processing session id " + sessionId);
+    @RequestMapping(path = "/pin", method = RequestMethod.GET)
+    public ResponseEntity<String> invite(@RequestParam String phoneNumber,
+                         @RequestHeader("Session-Id") String sessionId) throws NotificationClientException, UnsupportedEncodingException {
+ 	
+		if ( null == sessionId ) {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java]: Please switch these two variables around. sessionId should be compared to null and not the other way around (this is a readability issue more than anything).

File: src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java [hunk 3]
```diff
@@ -11,15 +16,37 @@
...
+			LOGGER.error("Session-Id request header not found");
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+		}
+    	LOGGER.info("Processing session id " + sessionId);
+    	
+		phoneNumber = URLDecoder.decode(phoneNumber, java.nio.charset.StandardCharsets.UTF_8.toString());
+		phoneNumber = phoneNumber.replaceAll("[ \\(\\)\\[\\]-]", "");			
+		
+		if ( !phoneNumber.matches("(\\+)*[0-9]+") ) {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java]: Please remove all unnecessary spaces - keep it consistent throughout the code base.

File: src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java (hunk 2 above)
Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java]: Also remove spaces here as well please.

File: src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java [hunk 4]
```diff
@@ -11,15 +16,37 @@
...
+    public ResponseEntity<String> invite(@RequestParam String phoneNumber,
+                         @RequestHeader("Session-Id") String sessionId) throws NotificationClientException, UnsupportedEncodingException {
+ 	
+		if ( null == sessionId ) {
+			LOGGER.error("Session-Id request header not found");
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+		}
+    	LOGGER.info("Processing session id " + sessionId);
+    	
+		phoneNumber = URLDecoder.decode(phoneNumber, java.nio.charset.StandardCharsets.UTF_8.toString());
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java]: Would be a lot more readable if we imported this class - java.nio.charset.StandardCharsets?

File: src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java [hunk 5]
```diff
@@ -11,15 +16,37 @@
...
+		phoneNumber = phoneNumber.replaceAll("[ \\(\\)\\[\\]-]", "");
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/probate/services/pin/controllers/PinController.java]: Could declare this regex as a constant at the top of the class.

**Comment by davejones74** [comment]
Thanks @tomxelliott changes have been done and I've tried to make the classes consistent as you've stated.

**Pull request closed without merging**